auto_changelog/__main__.py

```diff
@@ -11,6 +11,8 @@
     -o=OUTFILE --output=OUTFILE
                             The place to save the generated changelog 
                             [Default: CHANGELOG.md]
+    --latest-rev=LATEST_REV The latest git revision for generating changelog
+                            [Default: HEAD]
```

Comment:
Argparsing switch to click framework. So this has to be rewritten into it.

Also, I'm not sure about the `latest-rev` name. The revision we want to start from doesn't need to be the latest.
What about `from-revision`? This may be later coupled with `to-revision` so we can easily define changelog borders.

Reply:
Well, I agree with you.

Reply:
I Agree.

Reply:
Hi, 

I've added `--starting-commit` and `--stopping-commit` options for controlling the changelog borders and some other options that had been added when using `docopt`.

Since there's an other project with just the same name [auto-changelog](https://github.com/CookPete/auto-changelog) written in `nodejs`, which is also a well-known tool for generating changelog, which has an `--starting-commit` option working as the `from-revision`, I choose `--starting-commit` to rather than `from-revision` to keep the API compatibility proposed in https://github.com/Michael-F-Bryan/auto-changelog/issues/19.

Reply:
Great work, thank you.